middle_end/flambda/basic/continuation.ml

```diff
@@ -37,58 +41,92 @@ module Sort = struct
     | Toplevel_return
     | Exn
 
-  let to_int t =
+  let to_string t =
     match t with
-    | Normal -> 0
-    | Return -> 1
-    | Define_root_symbol -> 2
-    | Toplevel_return -> 3
-    | Exn -> 4
+    | Normal -> "Normal"
+    | Return -> "Return"
+    | Define_root_symbol -> "Define_root_symbol"
+    | Toplevel_return -> "Toplevel_return"
+    | Exn -> "Exn"
 
-  let of_int t =
-    match t with
-    | 0 -> Normal
-    | 1 -> Return
-    | 2 -> Define_root_symbol
-    | 3 -> Toplevel_return
-    | 4 -> Exn
-    | _ -> assert false
-
-  let num_bits_needed = 3
+  let print ppf t = Format.pp_print_string ppf (to_string t)
 end
 
-let max_num_continuations = (-1) lsr Sort.num_bits_needed
-let () = assert (max_num_continuations > 0)
+module Data = struct
+  type t = {
+    compilation_unit : Compilation_unit.t;
+    name : string;
+    name_stamp : int;
+    sort : Sort.t;
+  }
+
+  let flags = continuation_flags
+
+  let print ppf { compilation_unit; name; name_stamp; sort; } =
+    Format.fprintf ppf "@[<hov 1>(\
+        @[<hov 1>(compilation_unit@ %a)@]@ \
+        @[<hov 1>(name@ %s)@]@ \
+        @[<hov 1>(name_stamp@ %d)@]@ \
+        @[<hov 1>(sort@ %a)@]\
+        )@]"
+      Compilation_unit.print compilation_unit
+      name
+      name_stamp
+      Sort.print sort
+
+  let hash { compilation_unit; name = _; name_stamp; sort = _; } =
+    Hashtbl.hash (Compilation_unit.hash compilation_unit, name_stamp)
 
-let sort_shift = Sys.int_size - Sort.num_bits_needed
+  let equal t1 t2 =
+    if t1 == t2 then true
+    else
+      let { compilation_unit = compilation_unit1; name_stamp = name_stamp1; _
+          } = t1
```

Comment:
It's best not to use a catch-all underscore pattern in a situation like `equal`, as it's a place we really want to check if we change the type `t`.